src/lib389/lib389/monitor.py

```diff
@@ -86,8 +87,52 @@ def get_statistics(self):
         starttime = self.get_attr_vals_utf8('starttime')
         return (dtablesize, readwaiters, entriessent, bytessent, currenttime, starttime)
 
-    def get_status(self, use_json=False):
-        return self.get_attrs_vals_utf8([
+    def get_resource_stats(self):
+        """
+        Get CPU and memory stats
+        """
+        stats = {}
+        pid = self._instance.get_pid()
+        total_mem = psutil.virtual_memory()[0]
+        p = psutil.Process(pid)
+        memory_stats = p.memory_info()
+
+        # Get memory & CPU stats
+        stats['total_mem'] = [str(total_mem)]
+        stats['rss'] = [str(memory_stats[0])]
+        stats['vms'] = [str(memory_stats[1])]
+        stats['mem_percent'] = [str(int(p.memory_percent("vms")))]  # Round up
+        stats['total_threads'] = [str(p.num_threads())]
+        stats['cpu_usage'] = [str(int(p.cpu_percent()))]  # Round up
+
+        # Connections to DS
+        if self._instance.port == "0":
+            port = "ignore"
+        else:
+            port = str(self._instance.port)
+        if self._instance.sslport == "0":
+            sslport = "ignore"
+        else:
+            sslport = str(self._instance.sslport)
+
+        conn_count = 0
+        conns = psutil.net_connections()
+        for conn in conns:
+            if len(conn[4]) > 0:
+                conn_port = str(conn[4][1])
+                if conn_port in (port, sslport):
+                    conn_count += 1
+                    break
```

Comment:
Because the `break` it will always return `connection_count=1` or do I miss something?